test/cppsuite/test_harness/workload/thread_context.h

```diff
@@ -29,115 +29,66 @@
 #ifndef THREAD_CONTEXT_H
 #define THREAD_CONTEXT_H
 
-#include "../core/throttle.h"
+#include <string>
+
+extern "C" {
+#include "test_util.h"
+}
+
+#include "../core/configuration.h"
 #include "../timestamp_manager.h"
-#include "database_model.h"
-#include "random_generator.h"
+#include "../workload/database_model.h"
 #include "workload_tracking.h"
 
 namespace test_harness {
-class transaction_context {
-    public:
-    explicit transaction_context(configuration *config, timestamp_manager *timestamp_manager)
-        : _timestamp_manager(timestamp_manager)
-    {
-        /* Use optional here as our populate threads don't define this configuration. */
-        configuration *transaction_config = config->get_optional_subconfig(OPS_PER_TRANSACTION);
-        if (transaction_config != nullptr) {
-            _min_op_count = transaction_config->get_optional_int(MIN, 1);
-            _max_op_count = transaction_config->get_optional_int(MAX, 1);
-            delete transaction_config;
-        }
-    }
+enum thread_type { READ, INSERT, UPDATE };
 
-    /* Begin a transaction if we are not currently in one. */
-    void
-    try_begin(WT_SESSION *session, const std::string &config)
-    {
-        if (!_in_txn)
-            begin(session, config);
+static std::string
+type_string(thread_type type)
+{
+    switch (type) {
+    case thread_type::INSERT:
+        return ("insert");
+    case thread_type::READ:
+        return ("read");
+    case thread_type::UPDATE:
+        return ("update");
+    default:
+        testutil_die(EINVAL, "unexpected thread_type: %d", static_cast<int>(type));
     }
+}
 
-    void
-    begin(WT_SESSION *session, const std::string &config)
-    {
-        testutil_assert(!_in_txn);
-        testutil_check(
-          session->begin_transaction(session, config.empty() ? nullptr : config.c_str()));
-        /* This randomizes the number of operations to be executed in one transaction. */
-        _target_op_count =
-          random_generator::instance().generate_integer<int64_t>(_min_op_count, _max_op_count);
-        _op_count = 0;
-        _in_txn = true;
-    }
+class transaction_context {
+    public:
+    explicit transaction_context(configuration *config, timestamp_manager *timestamp_manager);
 
-    bool
-    active() const
-    {
-        return (_in_txn);
-    }
+    void begin(WT_SESSION *session, const std::string &config);
```

Comment:
I should've been specific can we order these functions:
```
active
add_op
begin
try_begin
commit
try_commit
rollback
try_rollback
```
And then do the same for the `.cxx`

Reply:
Done.